src/tests/blueprint/t_blueprint_mpi_verify.cpp

```diff
@@ -0,0 +1,149 @@
+TEST(blueprint_mpi_smoke, basic_verify)
+{
+    conduit::Node mesh, info;
+
+    // empty on all domains should return false ... 
+    EXPECT_FALSE( conduit::blueprint::mpi::verify("mesh",mesh,info, MPI_COMM_WORLD));
+        
+    conduit::blueprint::mesh::examples::braid("uniform",
+                                      10,
+                                      10,
+                                      10,
+                                      mesh);
+
+    EXPECT_TRUE( conduit::blueprint::mpi::verify("mesh",mesh,info, MPI_COMM_WORLD));
+    EXPECT_EQ(conduit::blueprint::mpi::mesh::number_of_domains(mesh,MPI_COMM_WORLD),2);
+}
+
+
+//-----------------------------------------------------------------------------
+TEST(blueprint_mpi_smoke, ranks_with_no_mesh)
+{
+    conduit::Node mesh, info;
+
+    // empty on all domains should return false ... 
+    EXPECT_FALSE( conduit::blueprint::mpi::verify("mesh",mesh,info, MPI_COMM_WORLD));
+
+    int par_rank;
+    int par_size;
+    MPI_Comm_rank(MPI_COMM_WORLD, &par_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &par_size);
+
+    // even with a single domain on one rank, we should still verify true
+    if(par_rank == 0)
+    {
+        conduit::blueprint::mesh::examples::braid("uniform",
+                                                  10,
+                                                  10,
+                                                  10,
+                                                  mesh);
+    }
+
+    EXPECT_TRUE( conduit::blueprint::mpi::verify("mesh",mesh,info, MPI_COMM_WORLD));
+
+    // check the number of domains
+    EXPECT_EQ( conduit::blueprint::mpi::mesh::number_of_domains(mesh,MPI_COMM_WORLD), 1);
+
+    // check hypothetical index gen
+    conduit::Node bp_index;
+    conduit::blueprint::mpi::mesh::generate_index(mesh,
+                                                  "",
+                                                  bp_index["mesh"],
+                                                  MPI_COMM_WORLD);
+
+
+    mesh.reset();
+    // even with a single domain on one rank, we should still verify true
+    if(par_rank == 1)
+    {
+        conduit::blueprint::mesh::examples::braid("uniform",
+                                                  10,
+                                                  10,
+                                                  10,
+                                                  mesh);
+    }
+
+    EXPECT_TRUE( conduit::blueprint::mpi::verify("mesh",mesh,info, MPI_COMM_WORLD) );
+
+    // check the number of domains
+    EXPECT_EQ( conduit::blueprint::mpi::mesh::number_of_domains(mesh,MPI_COMM_WORLD), 1);
+
+    // check hypothetical index gen
+    conduit::blueprint::mpi::mesh::generate_index(mesh,
+                                                  "",
+                                                  bp_index["mesh"],
+                                                  MPI_COMM_WORLD);
```

Comment:
Since there are two check procedures that are identical here (i.e. generate base mesh on some rank, verify, check domains, generate index), this should probably be a `for` loop instead to reduce redundancy.

Reply:
good suggestion, I'll implement